# Multilayer perceptron Neural net for county level medicare and COVID-19 data  
Author: Ting Fung Lam  
Predicts number of cases/deaths of covid-19 of a county

Reading data and  
Selecting only columns of interest from medicare_df and filling nulls

In [2]:
import pandas as pd

hospital_df = pd.read_csv("Hospitals_County_COVID_4.10.20.csv", thousands=',')
medicare_df = pd.read_csv("Medicare_2017.csv", thousands=',')
column_list = ['Zip_Code', 'NUM_HCPCS', 'NUM_Services', 'NUM_Medicare_BEN',
               'Total_Submitted_Charge_Amount', 'Total_Medicare_Allowed_Amount',
               'Total_Medicare_Payment_Amount',
               'Total_Medicare_Standardized_Payment_Amount',
               'NUM_HCPCS_Associated_With_Drug_Services', 'NUM_Drug_Services',
               'NUM_Medicare_BEN_With_Drug_Services',
               'Total_Drug_Submitted_Charge_Amount',
               'Total_Drug_Medicare_Allowed_Amount',
               'Total_Drug_Medicare_Payment_Amount',
               'Total_Drug_Medicare_Standardized_Payment_Amount',
               'NUM_HCPCS_Associated_With_Medical_Services', 'NUM_Medical_Services',
               'NUM_Medicare_BEN_With_Medical_Services',
               'Total_Medical_Submitted_Charge_Amount',
               'Total_Medical_Medicare_Allowed_Amount',
               'Total_Medical_Medicare_Payment_Amount',
               'Total_Medical_Medicare_Standardized_Payment_Amount',
               'Average_Age_of_BEN', 'NUM_BEN_Age_Less_65', 'NUM_BEN_Age_65_to_74',
               'NUM_BEN_Age_75_to_84', 'NUM_BEN_Age_Greater_84', 'NUM_Female_BEN',
               'NUM_Male_BEN', 'NUM_Non-Hispanic_White_BEN',
               'NUM_Black_or_African_American_BEN', 'NUM_Asian_Pacific_Islander_BEN',
               'NUM_Hispanic_BEN', 'NUM_American_IndianAlaska_Native_BEN',
               'NUM_BEN_With_Race_Not_Elsewhere_Classified',
               'NUM_BEN_With_Medicare_Only_Entitlement',
               'NUM_BEN_With_Medicare_Medicaid_Entitlement',
               'PCT_BEN_Atrial_Fibrillation', 'PCT_BEN_Alzheimers_Disease_or_Dementia',
               'PCT_BEN_Asthma', 'PCT_BEN_Cancer', 'PCT_BEN_Heart_Failure',
               'PCT_BEN_Chronic_Kidney_Disease',
               'PCT_BEN_Chronic_Obstructive_Pulmonary_Disease', 'PCT_BEN_Depression',
               'PCT_BEN_Diabetes', 'PCT_BEN_Hyperlipidemia', 'PCT_BEN_Hypertension',
               'PCT_BEN_Ischemic_Heart_Disease', 'PCT_BEN_Osteoporosis',
               'PCT_BEN_Rheumatoid_Arthritis_Osteoarthritis',
               'PCT_BEN_Schizophrenia_Other_Psychotic_Disorders', 'PCT_BEN_Stroke',
               'Average_HCC_Risk_Score_of_BEN']
medicare_df = medicare_df[column_list]
medicare_df = medicare_df.fillna(0)
medicare_df.head()

c:\users\administrator\appdata\local\programs\python\python38\lib\site-packages\IPython\core\interactiveshell.py:3062: DtypeWarning: Columns (10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Zip_Code  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0  215021854         23        1670.0               578   
1  602011718         20        8149.0              3703   
2  436233536         60        3085.0               278   
3  158251367         29         857.0               260   
4  441061716         29         155.0                71   

   Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                       800850.0                      177181.14   
1                      1272161.0                      319561.26   
2                       288873.0                      161220.25   
3                       132744.0                       88655.94   
4                        33700.0                       14572.28   

   Total_Medicare_Payment_Amount  Total_Medicare_Standardized_Payment_Amount  \
0                      137084.55                                   138279.12   
1                      243698.97                                   235917.60   
2                      119726.66                                   123488.10   
3                       66084.78                                    70643.97   
4                       11244.68                                    12001.69   

   NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                      0.0                0.0  ...   
1                                      0.0                0.0  ...   
2                                      2.0             1609.0  ...   
3                                      0.0                0.0  ...   
4                                      0.0                0.0  ...   

   PCT_BEN_Depression  PCT_BEN_Diabetes  PCT_BEN_Hyperlipidemia  \
0                46.0              51.0                    67.0   
1                12.0              20.0                    34.0   
2                45.0              41.0                    50.0   
3                50.0              44.0                    66.0   
4                21.0              20.0                    35.0   

   PCT_BEN_Hypertension  PCT_BEN_Ischemic_Heart_Disease  PCT_BEN_Osteoporosis  \
0                  75.0                            56.0                   9.0   
1                  55.0                            27.0                   9.0   
2                  74.0                            39.0                  10.0   
3                  75.0                            64.0                  20.0   
4                  48.0                            23.0                  15.0   

   PCT_BEN_Rheumatoid_Arthritis_Osteoarthritis  \
0                                         49.0   
1                                         36.0   
2                                         75.0   
3                                         48.0   
4                                         42.0   

   PCT_BEN_Schizophrenia_Other_Psychotic_Disorders  PCT_BEN_Stroke  \
0                                              8.0            13.0   
1                                              1.0             3.0   
2                                              4.0             5.0   
3                                             13.0            11.0   
4                                              0.0             0.0   

   Average_HCC_Risk_Score_of_BEN  
0                         2.1178  
1                         1.1024  
2                         1.6807  
3                         2.1173  
4                         1.0032  

[5 rows x 54 columns]

Formatting the Zip_Code column 

In [3]:
medicare_df['Zip_Code'] = medicare_df['Zip_Code'].astype(str).str.slice(stop=5)
medicare_df['Zip_Code'] = medicare_df.apply(pd.to_numeric, errors='coerce')
medicare_df = medicare_df.fillna(0)
medicare_df.head()

Zip_Code  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0   21502.0         23        1670.0               578   
1   60201.0         20        8149.0              3703   
2   43623.0         60        3085.0               278   
3   15825.0         29         857.0               260   
4   44106.0         29         155.0                71   

   Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                       800850.0                      177181.14   
1                      1272161.0                      319561.26   
2                       288873.0                      161220.25   
3                       132744.0                       88655.94   
4                        33700.0                       14572.28   

   Total_Medicare_Payment_Amount  Total_Medicare_Standardized_Payment_Amount  \
0                      137084.55                                   138279.12   
1                      243698.97                                   235917.60   
2                      119726.66                                   123488.10   
3                       66084.78                                    70643.97   
4                       11244.68                                    12001.69   

   NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                      0.0                0.0  ...   
1                                      0.0                0.0  ...   
2                                      2.0             1609.0  ...   
3                                      0.0                0.0  ...   
4                                      0.0                0.0  ...   

   PCT_BEN_Depression  PCT_BEN_Diabetes  PCT_BEN_Hyperlipidemia  \
0                46.0              51.0                    67.0   
1                12.0              20.0                    34.0   
2                45.0              41.0                    50.0   
3                50.0              44.0                    66.0   
4                21.0              20.0                    35.0   

   PCT_BEN_Hypertension  PCT_BEN_Ischemic_Heart_Disease  PCT_BEN_Osteoporosis  \
0                  75.0                            56.0                   9.0   
1                  55.0                            27.0                   9.0   
2                  74.0                            39.0                  10.0   
3                  75.0                            64.0                  20.0   
4                  48.0                            23.0                  15.0   

   PCT_BEN_Rheumatoid_Arthritis_Osteoarthritis  \
0                                         49.0   
1                                         36.0   
2                                         75.0   
3                                         48.0   
4                                         42.0   

   PCT_BEN_Schizophrenia_Other_Psychotic_Disorders  PCT_BEN_Stroke  \
0                                              8.0            13.0   
1                                              1.0             3.0   
2                                              4.0             5.0   
3                                             13.0            11.0   
4                                              0.0             0.0   

   Average_HCC_Risk_Score_of_BEN  
0                         2.1178  
1                         1.1024  
2                         1.6807  
3                         2.1173  
4                         1.0032  

[5 rows x 54 columns]

Compute sum and average for the respective columns of medicare table, group by Zip_Code  
Then put them back together by concat

In [4]:
sum = medicare_df.groupby(by='Zip_Code').sum().reset_index()
sum_columns = ['Zip_Code', 'NUM_HCPCS', 'NUM_Services', 'NUM_Medicare_BEN',
               'Total_Submitted_Charge_Amount', 'Total_Medicare_Allowed_Amount',
               'Total_Medicare_Payment_Amount',
               'Total_Medicare_Standardized_Payment_Amount',
               'NUM_HCPCS_Associated_With_Drug_Services', 'NUM_Drug_Services',
               'NUM_Medicare_BEN_With_Drug_Services',
               'Total_Drug_Submitted_Charge_Amount',
               'Total_Drug_Medicare_Allowed_Amount',
               'Total_Drug_Medicare_Payment_Amount',
               'Total_Drug_Medicare_Standardized_Payment_Amount',
               'NUM_HCPCS_Associated_With_Medical_Services', 'NUM_Medical_Services',
               'NUM_Medicare_BEN_With_Medical_Services',
               'Total_Medical_Submitted_Charge_Amount',
               'Total_Medical_Medicare_Allowed_Amount',
               'Total_Medical_Medicare_Payment_Amount',
               'Total_Medical_Medicare_Standardized_Payment_Amount','NUM_BEN_Age_Less_65', 'NUM_BEN_Age_65_to_74',
               'NUM_BEN_Age_75_to_84', 'NUM_BEN_Age_Greater_84', 'NUM_Female_BEN',
               'NUM_Male_BEN', 'NUM_Non-Hispanic_White_BEN',
               'NUM_Black_or_African_American_BEN', 'NUM_Asian_Pacific_Islander_BEN',
               'NUM_Hispanic_BEN', 'NUM_American_IndianAlaska_Native_BEN',
               'NUM_BEN_With_Race_Not_Elsewhere_Classified',
               'NUM_BEN_With_Medicare_Only_Entitlement',
               'NUM_BEN_With_Medicare_Medicaid_Entitlement',]
sum = sum[sum_columns]
sum.head()

Zip_Code  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0       0.0        497       13702.0              5457   
1       1.0         69         926.0               384   
2      80.0         12         195.0                77   
3      96.0          6          67.0                27   
4     115.0         18        1154.0               450   

   Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                     4220040.48                     1223773.71   
1                      381997.00                       92143.88   
2                       14742.98                       14616.36   
3                       11719.90                       11719.90   
4                      437354.00                      120603.99   

   Total_Medicare_Payment_Amount  Total_Medicare_Standardized_Payment_Amount  \
0                      918604.37                                   917482.14   
1                       70859.78                                    76989.29   
2                        9455.60                                    11451.76   
3                        8796.86                                     9888.65   
4                       92876.22                                    90738.11   

   NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                      2.0              278.0  ...   
1                                      0.0                0.0  ...   
2                                      0.0                0.0  ...   
3                                      0.0                0.0  ...   
4                                      0.0                0.0  ...   

   NUM_Female_BEN  NUM_Male_BEN  NUM_Non-Hispanic_White_BEN  \
0          3057.0        2336.0                      3960.0   
1           221.0         163.0                       367.0   
2             0.0           0.0                         0.0   
3             0.0           0.0                         0.0   
4           249.0         201.0                       414.0   

   NUM_Black_or_African_American_BEN  NUM_Asian_Pacific_Islander_BEN  \
0                              376.0                             0.0   
1                                0.0                             0.0   
2                                0.0                             0.0   
3                                0.0                             0.0   
4                                0.0                             0.0   

   NUM_Hispanic_BEN  NUM_American_IndianAlaska_Native_BEN  \
0             134.0                                  16.0   
1               0.0                                   0.0   
2               0.0                                   0.0   
3               0.0                                   0.0   
4              18.0                                   0.0   

   NUM_BEN_With_Race_Not_Elsewhere_Classified  \
0                                        34.0   
1                                         0.0   
2                                         0.0   
3                                         0.0   
4                                         0.0   

   NUM_BEN_With_Medicare_Only_Entitlement  \
0                                  3482.0   
1                                   324.0   
2                                    77.0   
3                                    27.0   
4                                   271.0   

   NUM_BEN_With_Medicare_Medicaid_Entitlement  
0                                      1938.0  
1                                        60.0  
2                                         0.0  
3                                         0.0  
4                                       179.0  

[5 rows x 36 columns]

In [5]:
average = medicare_df.groupby(by='Zip_Code').mean().reset_index()
average_columns = ['Average_Age_of_BEN', 'PCT_BEN_Atrial_Fibrillation', 'PCT_BEN_Alzheimers_Disease_or_Dementia',
               'PCT_BEN_Asthma', 'PCT_BEN_Cancer', 'PCT_BEN_Heart_Failure',
               'PCT_BEN_Chronic_Kidney_Disease',
               'PCT_BEN_Chronic_Obstructive_Pulmonary_Disease', 'PCT_BEN_Depression',
               'PCT_BEN_Diabetes', 'PCT_BEN_Hyperlipidemia', 'PCT_BEN_Hypertension',
               'PCT_BEN_Ischemic_Heart_Disease', 'PCT_BEN_Osteoporosis',
               'PCT_BEN_Rheumatoid_Arthritis_Osteoarthritis',
               'PCT_BEN_Schizophrenia_Other_Psychotic_Disorders', 'PCT_BEN_Stroke',]
average = average[average_columns]
average.head()

Average_Age_of_BEN  PCT_BEN_Atrial_Fibrillation  \
0                69.6                        11.12   
1                73.0                        23.00   
2                75.0                         0.00   
3                68.0                         0.00   
4                76.0                        38.00   

   PCT_BEN_Alzheimers_Disease_or_Dementia  PCT_BEN_Asthma  PCT_BEN_Cancer  \
0                                   21.32             9.4            9.16   
1                                   10.00             8.0           14.00   
2                                   18.00             0.0           16.00   
3                                   59.00             0.0            0.00   
4                                   38.00            17.0           18.00   

   PCT_BEN_Heart_Failure  PCT_BEN_Chronic_Kidney_Disease  \
0                  27.92                           42.16   
1                  32.00                           52.00   
2                  25.00                           55.00   
3                   0.00                            0.00   
4                  50.00                           69.00   

   PCT_BEN_Chronic_Obstructive_Pulmonary_Disease  PCT_BEN_Depression  \
0                                          24.72               40.88   
1                                          24.00               33.00   
2                                           0.00                0.00   
3                                           0.00               52.00   
4                                          37.00               51.00   

   PCT_BEN_Diabetes  PCT_BEN_Hyperlipidemia  PCT_BEN_Hypertension  \
0             36.64                   52.24                 68.52   
1             40.00                   56.00                 75.00   
2             61.00                   43.00                 75.00   
3             67.00                   56.00                 75.00   
4             36.00                   68.00                 75.00   

   PCT_BEN_Ischemic_Heart_Disease  PCT_BEN_Osteoporosis  \
0                           40.52                   5.8   
1                           44.00                  14.0   
2                           29.00                   0.0   
3                            0.00                   0.0   
4                           52.00                  23.0   

   PCT_BEN_Rheumatoid_Arthritis_Osteoarthritis  \
0                                        44.32   
1                                        50.00   
2                                        44.00   
3                                        67.00   
4                                        48.00   

   PCT_BEN_Schizophrenia_Other_Psychotic_Disorders  PCT_BEN_Stroke  
0                                             5.32            7.08  
1                                             0.00            7.00  
2                                             0.00            0.00  
3                                             0.00            0.00  
4                                            12.00           13.00

In [6]:
medicare_group_df = pd.concat([sum, average], axis=1)
medicare_group_df

Zip_Code  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0           0.0        497       13702.0              5457   
1           1.0         69         926.0               384   
2          80.0         12         195.0                77   
3          96.0          6          67.0                27   
4         115.0         18        1154.0               450   
...         ...        ...           ...               ...   
24756   99922.0         21         113.0                56   
24757   99925.0         52        1508.8               439   
24758   99926.0         60       71254.0               306   
24759   99928.0         87        1162.0                79   
24760   99929.0        107        2249.5               422   

       Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                         4220040.48                     1223773.71   
1                          381997.00                       92143.88   
2                           14742.98                       14616.36   
3                           11719.90                       11719.90   
4                          437354.00                      120603.99   
...                              ...                            ...   
24756                       16448.36                        6641.87   
24757                      423382.03                      105399.19   
24758                     5406780.03                     2096708.18   
24759                      121552.12                       39599.84   
24760                      284772.94                      126968.54   

       Total_Medicare_Payment_Amount  \
0                          918604.37   
1                           70859.78   
2                            9455.60   
3                            8796.86   
4                           92876.22   
...                              ...   
24756                        4855.14   
24757                       79956.90   
24758                     1576771.21   
24759                       29722.79   
24760                       90750.81   

       Total_Medicare_Standardized_Payment_Amount  \
0                                       917482.14   
1                                        76989.29   
2                                        11451.76   
3                                         9888.65   
4                                        90738.11   
...                                           ...   
24756                                     4583.19   
24757                                    75996.60   
24758                                  1654689.34   
24759                                    24645.37   
24760                                    79207.18   

       NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                          2.0              278.0  ...   
1                                          0.0                0.0  ...   
2                                          0.0                0.0  ...   
3                                          0.0                0.0  ...   
4                                          0.0                0.0  ...   
...                                        ...                ...  ...   
24756                                      0.0                0.0  ...   
24757                                      0.0                0.0  ...   
24758                                     28.0            66751.0  ...   
24759                                     28.0              448.0  ...   
24760                                      6.0               61.0  ...   

       PCT_BEN_Chronic_Obstructive_Pulmonary_Disease  PCT_BEN_Depression  \
0                                              24.72           40.880000   
1                                              24.00           33.000000   
2                                               0.00            0.000000   
3                                               0.00           52.000000   
4                     

COUNT the number of entries GROUP BY each zip code, and save to output_count.csv

In [29]:
count = medicare_df.iloc[:, 0:2].groupby(by='Zip_Code').count().reset_index()
count.to_csv('output_count.csv')
count

Zip_Code  NUM_HCPCS
0           0.0         25
1           1.0          1
2          80.0          1
3          96.0          1
4         115.0          1
...         ...        ...
24756   99922.0          2
24757   99925.0          4
24758   99926.0          3
24759   99928.0          1
24760   99929.0          5

[24761 rows x 2 columns]

INNER JOIN the hospital and medicare tables ON Zip_Code

In [7]:
merge_df = pd.merge(medicare_df, hospital_df, how='inner', left_on='Zip_Code', right_on='ZIP')
merge_df.head()

Zip_Code  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0   21502.0         23        1670.0               578   
1   21502.0         23        1670.0               578   
2   21502.0          8        1827.0               601   
3   21502.0          8        1827.0               601   
4   21502.0         49        8450.0              1029   

   Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                      800850.00                      177181.14   
1                      800850.00                      177181.14   
2                     1282363.64                      187382.74   
3                     1282363.64                      187382.74   
4                     2379228.00                      881333.53   

   Total_Medicare_Payment_Amount  Total_Medicare_Standardized_Payment_Amount  \
0                      137084.55                                   138279.12   
1                      137084.55                                   138279.12   
2                      145298.83                                   142633.13   
3                      145298.83                                   142633.13   
4                      719342.57                                   717252.38   

   NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                      0.0                0.0  ...   
1                                      0.0                0.0  ...   
2                                      0.0                0.0  ...   
3                                      0.0                0.0  ...   
4                                      0.0                0.0  ...   

   COUNTYFIPS    COUNTY                           HOSPITAL_DESC  \
0       24001  ALLEGANY  GENERAL MEDICAL AND SURGICAL HOSPITALS   
1       24001  ALLEGANY  GENERAL MEDICAL AND SURGICAL HOSPITALS   
2       24001  ALLEGANY  GENERAL MEDICAL AND SURGICAL HOSPITALS   
3       24001  ALLEGANY  GENERAL MEDICAL AND SURGICAL HOSPITALS   
4       24001  ALLEGANY  GENERAL MEDICAL AND SURGICAL HOSPITALS   

                OWNER  BEDS         TRAUMA  HELIPAD  POPULATION_COUNTY  \
0          NON-PROFIT   371      LEVEL III        Y              70416   
1  GOVERNMENT - STATE    80  NOT AVAILABLE        N              70416   
2          NON-PROFIT   371      LEVEL III        Y              70416   
3  GOVERNMENT - STATE    80  NOT AVAILABLE        N              70416   
4          NON-PROFIT   371      LEVEL III        Y              70416   

   COVID_CASES_4.10.20  COVID_DEATHS_4.10.20  
0                   10                     0  
1                   10                     0  
2                   10                     0  
3                   10                     0  
4                   10                     0  

[5 rows x 75 columns]

Repeat sum/average GROUP BY COUNTYFIPS

In [8]:
sum = merge_df.groupby(by='COUNTYFIPS').sum().reset_index()
sum_columns = ['COUNTYFIPS', 'NUM_HCPCS', 'NUM_Services', 'NUM_Medicare_BEN',
               'Total_Submitted_Charge_Amount', 'Total_Medicare_Allowed_Amount',
               'Total_Medicare_Payment_Amount',
               'Total_Medicare_Standardized_Payment_Amount',
               'NUM_HCPCS_Associated_With_Drug_Services', 'NUM_Drug_Services',
               'NUM_Medicare_BEN_With_Drug_Services',
               'Total_Drug_Submitted_Charge_Amount',
               'Total_Drug_Medicare_Allowed_Amount',
               'Total_Drug_Medicare_Payment_Amount',
               'Total_Drug_Medicare_Standardized_Payment_Amount',
               'NUM_HCPCS_Associated_With_Medical_Services', 'NUM_Medical_Services',
               'NUM_Medicare_BEN_With_Medical_Services',
               'Total_Medical_Submitted_Charge_Amount',
               'Total_Medical_Medicare_Allowed_Amount',
               'Total_Medical_Medicare_Payment_Amount',
               'Total_Medical_Medicare_Standardized_Payment_Amount','NUM_BEN_Age_Less_65', 'NUM_BEN_Age_65_to_74',
               'NUM_BEN_Age_75_to_84', 'NUM_BEN_Age_Greater_84', 'NUM_Female_BEN',
               'NUM_Male_BEN', 'NUM_Non-Hispanic_White_BEN',
               'NUM_Black_or_African_American_BEN', 'NUM_Asian_Pacific_Islander_BEN',
               'NUM_Hispanic_BEN', 'NUM_American_IndianAlaska_Native_BEN',
               'NUM_BEN_With_Race_Not_Elsewhere_Classified',
               'NUM_BEN_With_Medicare_Only_Entitlement',
               'NUM_BEN_With_Medicare_Medicaid_Entitlement',]
sum = sum[sum_columns]
sum.head()

COUNTYFIPS  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0        1001        649       23608.4              5744   
1        1003      17736     1602538.8            162262   
2        1005        738       93074.1              8758   
3        1007        539       30371.2              3917   
4        1009        656       69980.4              6690   

   Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                   5.642743e+06                     1601574.96   
1                   1.515522e+08                    55396151.32   
2                   8.347437e+06                     3250431.51   
3                   2.129942e+06                     1155227.84   
4                   5.193917e+06                     2358346.25   

   Total_Medicare_Payment_Amount  Total_Medicare_Standardized_Payment_Amount  \
0                     1211002.80                                  1322271.14   
1                    41633885.71                                 44788496.62   
2                     2436659.90                                  2543344.20   
3                      862891.22                                   891117.87   
4                     1774046.43                                  1766102.27   

   NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                     80.0             2511.0  ...   
1                                   1438.0           918468.0  ...   
2                                     60.0             3336.0  ...   
3                                     31.0              451.0  ...   
4                                     93.0             3023.0  ...   

   NUM_Female_BEN  NUM_Male_BEN  NUM_Non-Hispanic_White_BEN  \
0          3228.0        2484.0                      3753.0   
1         90735.0       69942.0                    132868.0   
2          5251.0        3507.0                      5335.0   
3          2237.0        1489.0                      2634.0   
4          3987.0        2690.0                      4152.0   

   NUM_Black_or_African_American_BEN  NUM_Asian_Pacific_Islander_BEN  \
0                              690.0                             0.0   
1                             7786.0                           146.0   
2                             1134.0                             0.0   
3                               11.0                             0.0   
4                               14.0                             0.0   

   NUM_Hispanic_BEN  NUM_American_IndianAlaska_Native_BEN  \
0               0.0                                   0.0   
1             578.0                                 274.0   
2              28.0                                   0.0   
3               0.0                                   0.0   
4               0.0                                   0.0   

   NUM_BEN_With_Race_Not_Elsewhere_Classified  \
0                                         0.0   
1                                       529.0   
2                                         0.0   
3                                         0.0   
4                                         0.0   

   NUM_BEN_With_Medicare_Only_Entitlement  \
0                                  4159.0   
1                                128890.0   
2                                  5359.0   
3                                  2203.0   
4                                  4280.0   

   NUM_BEN_With_Medicare_Medicaid_Entitlement  
0                                      1345.0  
1                                     21063.0  
2                                      3359.0  
3                                      1404.0  
4                                      2229.0  

[5 rows x 36 columns]

In [9]:
average = merge_df.groupby(by='COUNTYFIPS').mean().reset_index()
average_columns = ['Average_Age_of_BEN', 'PCT_BEN_Atrial_Fibrillation', 'PCT_BEN_Alzheimers_Disease_or_Dementia',
               'PCT_BEN_Asthma', 'PCT_BEN_Cancer', 'PCT_BEN_Heart_Failure',
               'PCT_BEN_Chronic_Kidney_Disease',
               'PCT_BEN_Chronic_Obstructive_Pulmonary_Disease', 'PCT_BEN_Depression',
               'PCT_BEN_Diabetes', 'PCT_BEN_Hyperlipidemia', 'PCT_BEN_Hypertension',
               'PCT_BEN_Ischemic_Heart_Disease', 'PCT_BEN_Osteoporosis',
               'PCT_BEN_Rheumatoid_Arthritis_Osteoarthritis',
               'PCT_BEN_Schizophrenia_Other_Psychotic_Disorders', 'PCT_BEN_Stroke',]
average = average[average_columns]
average.head()

Average_Age_of_BEN  PCT_BEN_Atrial_Fibrillation  \
0           71.076923                    11.769231   
1           71.638202                    10.970787   
2           70.928571                     8.642857   
3           70.000000                     8.666667   
4           71.538462                     7.346154   

   PCT_BEN_Alzheimers_Disease_or_Dementia  PCT_BEN_Asthma  PCT_BEN_Cancer  \
0                               13.692308        5.307692        9.461538   
1                               12.038202        6.020225        9.164045   
2                               14.750000        6.607143        8.642857   
3                               16.033333        4.466667        4.366667   
4                               16.038462        5.615385        4.692308   

   PCT_BEN_Heart_Failure  PCT_BEN_Chronic_Kidney_Disease  \
0              23.807692                       36.307692   
1              16.923596                       28.525843   
2              17.892857                       33.428571   
3              20.133333                       34.866667   
4              18.769231                       28.576923   

   PCT_BEN_Chronic_Obstructive_Pulmonary_Disease  PCT_BEN_Depression  \
0                                      19.230769           21.653846   
1                                      14.817978           25.094382   
2                                      14.928571           23.892857   
3                                      14.966667           21.400000   
4                                      19.115385           27.384615   

   PCT_BEN_Diabetes  PCT_BEN_Hyperlipidemia  PCT_BEN_Hypertension  \
0         34.923077               57.500000             68.500000   
1         28.071910               50.604494             67.505618   
2         36.928571               56.035714             73.178571   
3         30.500000               44.266667             68.933333   
4         32.423077               58.307692             73.384615   

   PCT_BEN_Ischemic_Heart_Disease  PCT_BEN_Osteoporosis  \
0                       40.076923              5.769231   
1                       36.462921              6.233708   
2                       31.714286              3.857143   
3                       28.733333              5.800000   
4                       34.230769              5.115385   

   PCT_BEN_Rheumatoid_Arthritis_Osteoarthritis  \
0                                    45.000000   
1                                    49.602247   
2                                    49.785714   
3                                    41.566667   
4                                    40.769231   

   PCT_BEN_Schizophrenia_Other_Psychotic_Disorders  PCT_BEN_Stroke  
0                                         4.115385        6.923077  
1                                         3.856180        4.265169  
2                                         6.071429        5.500000  
3                                         3.766667        5.533333  
4                                         4.000000        5.038462

In [10]:
merge_df = pd.concat([sum, average], axis=1)
merge_df

COUNTYFIPS  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0           1001        649       23608.4              5744   
1           1003      17736     1602538.8            162262   
2           1005        738       93074.1              8758   
3           1007        539       30371.2              3917   
4           1009        656       69980.4              6690   
...          ...        ...           ...               ...   
2525       56037       3118      180247.3             24430   
2526       56039       3707      123737.1             19474   
2527       56041       1305       52681.9              8436   
2528       56043        451       25644.3              3605   
2529       56045        277       16037.0              1847   

      Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                      5.642743e+06                     1601574.96   
1                      1.515522e+08                    55396151.32   
2                      8.347437e+06                     3250431.51   
3                      2.129942e+06                     1155227.84   
4                      5.193917e+06                     2358346.25   
...                             ...                            ...   
2525                   4.199070e+07                     9361492.71   
2526                   1.759663e+07                     5335013.71   
2527                   7.084807e+06                     2634476.75   
2528                   2.650999e+06                     1204296.03   
2529                   6.767309e+05                      374177.73   

      Total_Medicare_Payment_Amount  \
0                        1211002.80   
1                       41633885.71   
2                        2436659.90   
3                         862891.22   
4                        1774046.43   
...                             ...   
2525                     6968556.74   
2526                     3964972.08   
2527                     1941228.35   
2528                      858835.18   
2529                      278743.70   

      Total_Medicare_Standardized_Payment_Amount  \
0                                     1322271.14   
1                                    44788496.62   
2                                     2543344.20   
3                                      891117.87   
4                                     1766102.27   
...                                          ...   
2525                                  6774301.39   
2526                                  3984418.68   
2527                                  1923237.45   
2528                                   860269.73   
2529                                   232141.73   

      NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                        80.0             2511.0  ...   
1                                      1438.0           918468.0  ...   
2                                        60.0             3336.0  ...   
3                                        31.0              451.0  ...   
4                                        93.0             3023.0  ...   
...                                       ...                ...  ...   
2525                                    157.0             5408.0  ...   
2526                                    111.0            54315.0  ...   
2527                                     51.0             1456.0  ...   
2528                                     49.0             3797.0  ...   
2529                                      7.0              141.0  ...   

      PCT_BEN_Chronic_Obstructive_Pulmonary_Disease  PCT_BEN_Depression  \
0                                         19.230769           21.653846   
1                                         14.817978           25.094382   
2                                         14.928571           23.892857   
3                                         14.966667           21.400000   
4                                         19.115385           

Extracting county data from hospital data

In [11]:
column_list = ['COUNTYFIPS', 'POPULATION_COUNTY', 'COVID_CASES_4.10.20', 'COVID_DEATHS_4.10.20']
hospital_df = hospital_df[column_list].drop_duplicates(column_list)
hospital_df

COUNTYFIPS  POPULATION_COUNTY  COVID_CASES_4.10.20  COVID_DEATHS_4.10.20
0          30079               1077                    0                     0
1          20071               1232                    0                     0
2          30011               1252                    0                     0
3          30033               1258                    0                     0
4          31149               1357                    0                     0
...          ...                ...                  ...                   ...
7491       46043               2921                    0                     0
7492       46045               3829                    0                     0
7494       37003              37497                    2                     0
7495       46031               4086                    1                     0
7496       51127              23091                   13                     0

[2554 rows x 4 columns]

In [12]:
final_df = pd.merge(merge_df, hospital_df, on='COUNTYFIPS')
final_df

COUNTYFIPS  NUM_HCPCS  NUM_Services  NUM_Medicare_BEN  \
0           1001        649       23608.4              5744   
1           1003      17736     1602538.8            162262   
2           1005        738       93074.1              8758   
3           1007        539       30371.2              3917   
4           1009        656       69980.4              6690   
...          ...        ...           ...               ...   
2525       56037       3118      180247.3             24430   
2526       56039       3707      123737.1             19474   
2527       56041       1305       52681.9              8436   
2528       56043        451       25644.3              3605   
2529       56045        277       16037.0              1847   

      Total_Submitted_Charge_Amount  Total_Medicare_Allowed_Amount  \
0                      5.642743e+06                     1601574.96   
1                      1.515522e+08                    55396151.32   
2                      8.347437e+06                     3250431.51   
3                      2.129942e+06                     1155227.84   
4                      5.193917e+06                     2358346.25   
...                             ...                            ...   
2525                   4.199070e+07                     9361492.71   
2526                   1.759663e+07                     5335013.71   
2527                   7.084807e+06                     2634476.75   
2528                   2.650999e+06                     1204296.03   
2529                   6.767309e+05                      374177.73   

      Total_Medicare_Payment_Amount  \
0                        1211002.80   
1                       41633885.71   
2                        2436659.90   
3                         862891.22   
4                        1774046.43   
...                             ...   
2525                     6968556.74   
2526                     3964972.08   
2527                     1941228.35   
2528                      858835.18   
2529                      278743.70   

      Total_Medicare_Standardized_Payment_Amount  \
0                                     1322271.14   
1                                    44788496.62   
2                                     2543344.20   
3                                      891117.87   
4                                     1766102.27   
...                                          ...   
2525                                  6774301.39   
2526                                  3984418.68   
2527                                  1923237.45   
2528                                   860269.73   
2529                                   232141.73   

      NUM_HCPCS_Associated_With_Drug_Services  NUM_Drug_Services  ...  \
0                                        80.0             2511.0  ...   
1                                      1438.0           918468.0  ...   
2                                        60.0             3336.0  ...   
3                                        31.0              451.0  ...   
4                                        93.0             3023.0  ...   
...                                       ...                ...  ...   
2525                                    157.0             5408.0  ...   
2526                                    111.0            54315.0  ...   
2527                                     51.0             1456.0  ...   
2528                                     49.0             3797.0  ...   
2529                                      7.0              141.0  ...   

      PCT_BEN_Hyperlipidemia  PCT_BEN_Hypertension  \
0                  57.500000             68.500000   
1                  50.604494             67.505618   
2                  56.035714             73.178571   
3                  44.266667             68.933333   
4                  58.307692             73.384615   
...                      ...                   ...   
2525               22.184466             48.456311   
2526 


## Neural Net  
Defining np arrays of features X and labels y 

In [13]:
X = final_df.loc[:, 'NUM_HCPCS':'POPULATION_COUNTY'].values
y = final_df['COVID_CASES_4.10.20'].values

Training and testing Multilayer perceptron using Keras  
Outputs Mean squared error and mean absolute error at the end

In [14]:
import tensorflow as tf
from tensorflow.keras import models, layers
from tensorflow.keras.layers import Dropout
from tensorflow.keras import utils
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
scaler = StandardScaler().fit(X_train)
x_train_scaled = scaler.transform(X_train)
x_test_scaled = scaler.transform(X_test)

model = models.Sequential()
model.add(layers.Dense(64, activation=tf.nn.relu, input_dim=X_train.shape[1]))
model.add(layers.Dense(64, activation=tf.nn.relu))
model.add(layers.Dense(32, activation=tf.nn.relu))
model.add(layers.Dense(16, activation=tf.nn.relu))
model.add(layers.Dense(1))


model.compile(optimizer='rmsprop', loss='mse', metrics=['mae'])
model.fit(x_train_scaled, y_train, validation_split=0.2, epochs=100)

model.evaluate(x_test_scaled, y_test)

Epoch 1/100
51/51 [==============================] - 0s 9ms/step - loss: 998888.0625 - mae: 154.9958 - val_loss: 3354156.0000 - val_mae: 259.0067
Epoch 2/100
51/51 [==============================] - 0s 5ms/step - loss: 1114642.8750 - mae: 147.4221 - val_loss: 3264554.5000 - val_mae: 256.4597
Epoch 3/100
51/51 [==============================] - 0s 5ms/step - loss: 861886.1250 - mae: 150.8011 - val_loss: 3194457.0000 - val_mae: 256.9416
Epoch 4/100
51/51 [==============================] - 0s 6ms/step - loss: 862173.7500 - mae: 159.3073 - val_loss: 3136455.5000 - val_mae: 258.5631
Epoch 5/100
51/51 [==============================] - 0s 6ms/step - loss: 836153.3125 - mae: 158.4624 - val_loss: 3117852.2500 - val_mae: 256.1879
Epoch 6/100
51/51 [==============================] - 0s 5ms/step - loss: 826734.4375 - mae: 162.8887 - val_loss: 3115519.5000 - val_mae: 250.5932
Epoch 7/100
51/51 [==============================] - 0s 5ms/step - loss: 822913.9375 - mae: 160.6231 - val_loss: 3092507.50

[592448.0625, 158.16250610351562]

In conclusion, the neural net gives Mean Absolute error of 158 cases and Mean squared error of around 600,000
